src/app/Scenes/SavedSearchAlert/screens/AddFiltersScreenRarity.tsx

```diff
@@ -0,0 +1,68 @@
+import { Flex, Spacer, Text } from "@artsy/palette-mobile"
+import { ATTRIBUTION_CLASS_OPTIONS } from "app/Components/ArtworkFilter/Filters/AttributionClassOptions"
+import { SearchCriteria } from "app/Components/ArtworkFilter/SavedSearch/types"
+import { SavedSearchStore } from "app/Scenes/SavedSearchAlert/SavedSearchStore"
+import { isValueSelected, useSearchCriteriaAttributes } from "app/Scenes/SavedSearchAlert/helpers"
+import { FillPill } from "app/Scenes/SavedSearchAlert/screens/AddFilterPill"
+
+export const AddFiltersScreenRarity = () => {
+  const selectedAttributes = useSearchCriteriaAttributes(
+    SearchCriteria.attributionClass
+  ) as string[]
+
+  const setValueToAttributesByKeyAction = SavedSearchStore.useStoreActions(
+    (actions) => actions.setValueToAttributesByKeyAction
+  )
+  const removeValueFromAttributesByKeyAction = SavedSearchStore.useStoreActions(
+    (actions) => actions.removeValueFromAttributesByKeyAction
+  )
+
+  const handlePress = (value: string) => {
+    if (
+      isValueSelected({
+        selectedAttributes,
+        value: value,
+      })
+    ) {
+      removeValueFromAttributesByKeyAction({
+        key: SearchCriteria.attributionClass,
+        value: value,
+      })
+    } else {
+      const newValues = (selectedAttributes || []).concat(value)
+
+      setValueToAttributesByKeyAction({
+        key: SearchCriteria.attributionClass,
+        value: newValues,
+      })
+    }
+  }
```

Comment:
This might be easier to read 🤷 

```suggestion
  const handlePress = (value: string) => {
    const isSelected = isValueSelected({
      selectedAttributes,
      value: value,
    })

    if (isSelected) {
      removeValueFromAttributesByKeyAction({
        key: SearchCriteria.attributionClass,
        value: value,
      })
    } else {
      const newValues = (selectedAttributes || []).concat(value)

      setValueToAttributesByKeyAction({
        key: SearchCriteria.attributionClass,
        value: newValues,
      })
    }
  }
```

Reply:
agree